tests/playwright-test/reporter-html.spec.ts

```diff
@@ -473,6 +474,37 @@ test('should show multi trace source', async ({ runInlineTest, page, server, sho
   await expect(page.locator('.source-line-running')).toContainText('request.get');
 });
 
+test('should warn user when viewing via file:// protocol', async ({ runInlineTest, page, showReport }, testInfo) => {
```

Comment:
Added tests since—it would be pretty unexpected if the dialog showed up when it wasn't supposed to!